engine/crates/composition/src/subgraphs/field_types.rs

```diff
@@ -120,3 +172,22 @@ impl<'a> FieldTypeWalker<'a> {
             .unwrap_or_else(|| self.inner().is_required)
     }
 }
+
+impl std::fmt::Display for FieldTypeWalker<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut out = self.type_name().as_str().to_owned();
```

Comment:
Lots of allocations, but it's only for errors so I don't think we should care, it doesn't slow down the happy path.